src/text/writer.rs

```diff
@@ -290,16 +286,16 @@ impl<W: Write> TextWriter<W> {
     }
 
     /// Writes the provided byte array slice as an Ion clob.
-    pub fn write_clob(&mut self, _value: &[u8]) -> IonResult<()> {
+    pub fn write_clob(&mut self, value: &[u8]) -> IonResult<()> {
         // clob_value to be written based on defined STRING_ESCAPE_CODES.
-        let mut clob_value = String::from("");
-        for i in 0.._value.len() {
-            let c = (_value[i] & 0xff) as char;
+        let mut clob_value = String::new();
```

Comment:
We can save ourselves a few reallocations by making an educated guess at how long `clob_value` will be and using [`String::with_capacity`](https://doc.rust-lang.org/std/string/struct.String.html#method.with_capacity) to set aside that much memory.

In the best case, each byte from `value` will be displayable ASCII and so will add one byte to `clob_value`. In the worst case, each byte from `value` will be non-ASCII and will add 4 bytes (`\xHH`) to `clob_value`. (Escaped ASCII values like `\n` add two bytes and are neither the best nor worst case.)

In any case, we also have to add enough space for the `{{` and `}}`, 4 more bytes.

To be very conservative, we can start by accommodating the worst case:
```suggestion
        const NUM_DELIMITER_BYTES: usize = 4; // {{}}
        const NUM_HEX_BYTES_PER_BYTE: usize = 4; // \xHH
        // Set aside enough memory to hold a clob containing all hex-encoded bytes
        let mut clob_value = String::with_capacity((value.len() * NUM_HEX_BYTES_PER_BYTE) + NUM_DELIMITER_BYTES);
```

We could also go with a middle-ground approach that multiplied `value.len()` by 1, 2, or 3 instead, occasionally requiring reallocations but wasting less memory in the common case.